src/constants.ts

```diff
@@ -1,9 +1,30 @@
 import { ClientAuthMethod, ResponseType } from 'openid-client'
 
+/**
+ * sgID Defaults
+ */
 export const SGID_SIGNING_ALG = 'RS256'
-export const DEFAULT_SGID_CODE_CHALLENGE_METHOD = 'S256'
 export const DEFAULT_SCOPE = 'myinfo.nric_number openid'
 export const SGID_SUPPORTED_FLOWS: ResponseType[] = ['code']
 export const SGID_AUTH_METHOD: ClientAuthMethod = 'client_secret_post'
 
+/**
+ * PKCE Defaults
+ */
+export const DEFAULT_SGID_CODE_CHALLENGE_METHOD = 'S256'
+
+/**
+ * API Versioning
+ */
 export const API_VERSION = 2
+
+/**
+ * Testing
+ */
+/**
+ * Regex pattern that the code verifier and code challenge in the PKCE flow should match according to the PKCE RFC
+ * https://www.rfc-editor.org/rfc/rfc7636
+ */
+export const codeVerifierAndChallengePattern = /^[A-Za-z\d\-._~]{43,128}$/
```

Comment:
is this used purely in test code? if so, we should put it together with the test code instead of bundling it together with SDK code. it's good to be deliberate about what we export, as we have to support it forever or make a breaking change to remove it

Reply:
mm that's a very good point - shifted into the test code in this [PR ](https://github.com/opengovsg/sgid-client/pull/35)